File: server/app/controllers/WsCaisseDepartement.java
package controllers;

import utils.dao.CaisseDepartementaleDao;
import utils.engine.data.enums.ChecklistName;

public class WsCaisseDepartement extends SecuredRetraiteController {

	public static void deleteDepartement(final Long caisseId, final String departement) {
		new CaisseDepartementaleDao().deleteDepartement(caisseId, departement);
	}

	public static void addDepartement(final ChecklistName checklistName, final Long caisseId, final String departement) {
		new CaisseDepartementaleDao().addDepartement(checklistName, caisseId, departement);
	}
}

File: client/app/src/config/data/ws/ws-caisse-departement/ws-caisse-departement.js
'use strict';

angular.module('SgmapRetraiteConfig').service('WsCaisseDepartement', function ($q, $http) {
 
    this.deleteDepartement = function(caisseId, departement) {
        return callWs('deletedepartement', caisseId, departement);
    };
    
    this.addDepartement = function(checklistName, caisseId, departement) {
        return callWs('adddepartement', caisseId, departement, checklistName);
    };
    
    // Privé
    
    function callWs(operation, caisseId, departement, checklistName) {
        var url = '/ws/caissedepartement/'+operation+'?';
        if (checklistName) {
            url += 'checklistName='+checklistName+'&';
        }
        url += 'caisseId='+caisseId+'&departement='+departement;
        
        var deferred = $q.defer();
        $http.get(url)
            .then(function() {
                deferred.resolve();
            }, function onError(error) {
                deferred.reject(error);
            });
        return deferred.promise;
    }
});
